lib/SimpleSAML/Utils/Crypto.php

```diff
@@ -356,6 +356,9 @@ public static function secureCompare(string $known, string $user): bool
      */
     public static function pwValid(string $hash, string $password): bool
     {
+        if (!is_null(password_get_info($password)['algo'])) {
+            throw new Error\Exception("Cannot use a hash value for authentication.");
+        }
         if (password_verify($password, $hash)) {
```

Comment:
Wouldn't it be sufficient to reduce this entire method to `return password_verify($password, $hash);` ?

Reply:
Ah, I had missed your comment in the issue... So this fix would make sense in 1.19, but for master I think it should be reduced to the oneliner above

Reply:
That depends if we indeed want to force users to generate a hash for this (so disallow plain text). It does make initial setup quite a bit less straightforward.

Reply:
I will merge this because it improves the situation and we can discuss making the plain text version obsolete in a separate PR.